### PR TITLE
MAINTAINERS: Add BeagleBoard Platforms

### PR DESCRIPTION
**Pull request by Ayush1325**
- Due to the incresing support for ZephyrRTOS in BeagleBoard boards, it would be good to have this entry.
- Was discussed in Discord when asking for help for PR review

Details about BealgeBoard boards can be found here:
- https://www.beagleboard.org
- https://docs.beagleboard.org/latest/

cc @cfriedt

### CODE REVIEW

File: MAINTAINERS.yml [hunk 1]
```diff
@@ -4618,3 +4618,17 @@ zbus:
...
+    - jadonk
+  collaborators:
+    - ayush1325
+    - vaishnavachath
+    - con-pax
+  files:
+    - boards/arm/beagle_bcf/
+    - boards/riscv/beaglev_fire/
+  labels:
+    - "area: BeagleBOard Platforms"
```
Comment (kartben) [inline MAINTAINERS.yml]:
```suggestion
    - "area: BeagleBoard Platforms"
```

**Review by cfriedt — COMMENTED** [review]
I think there was an effort (at least at some time) to keep entries in alphabetical order. Can this be moved up to the right location?

File: MAINTAINERS.yml [hunk 2]
```diff
@@ -259,6 +259,20 @@ Ambiq Platforms:
...
+    - jadonk
+  collaborators:
+    - ayush1325
+    - vaishnavachath
+    - con-pax
+  files:
+    - boards/arm/beagle_bcf/
+    - boards/riscv/beaglev_fire/
+  labels:
+    - "area: BeagleBoard Platforms"
```
Comment (kartben) [inline MAINTAINERS.yml:274]:
```suggestion
    - "platform: BeagleBoard"
```

File: MAINTAINERS.yml [hunk 3]
```diff
@@ -259,6 +259,20 @@ Ambiq Platforms:
...
+  status: maintained
+  maintainers:
+    - jadonk
+  collaborators:
+    - ayush1325
+    - vaishnavachath
+    - con-pax
+  files:
+    - boards/arm/beagle_bcf/
+    - boards/riscv/beaglev_fire/
```
Comment (kartben) [inline MAINTAINERS.yml:272]: probably doesn't hurt to do it this way:

```suggestion
    - boards/arm/beagle*
    - boards/riscv/beagle*
```

**Review by kartben — CHANGES_REQUESTED** [review]
also, alphabetical order like Chris mentioned. Thanks!

**Comment by Ayush1325** [comment]
Well, not sure why that is failing.

File: MAINTAINERS.yml [hunk 4]
```diff
@@ -259,6 +259,20 @@ Ambiq Platforms:
...
+    - jadonk
+  collaborators:
+    - ayush1325
+    - con-pax
+    - vaishnavachath
+  files:
+    - boards/arm/beagle*
+    - boards/riscv/beagle*
+  labels:
+    - "platform: BeagleBoard Platforms"
```
Comment (kartben) [inline MAINTAINERS.yml:274]: don't think any other platform label has "... Platforms" in them, hence why my previous suggestion :) 

```suggestion
    - "platform: BeagleBoard"
```

File: MAINTAINERS.yml [hunk 5]
```diff
@@ -259,6 +259,20 @@ Ambiq Platforms:
...
+  status: maintained
+  maintainers:
+    - jadonk
+  collaborators:
+    - ayush1325
+    - con-pax
+    - vaishnavachath
+  files:
+    - boards/arm/beagle*
+    - boards/riscv/beagle*
```
Comment (kartben) [inline MAINTAINERS.yml:272]: failure is my fault, sorry, should have a trailing slash

```suggestion
    - boards/arm/beagle*/
    - boards/riscv/beagle*/
```

**Comment by jadonk** [comment]
Do you need a signed-off-by from me? Do you have the removal of the term "Platforms"?

**Review by kartben — APPROVED** [review]

**Comment by kartben** [comment]
> Do you need a signed-off-by from me? Do you have the removal of the term "Platforms"?

I do not think so, your +1 here should be more than enough

**Review by cfriedt — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Comment by kartben** [comment]
@jadonk thanks for stepping up as a maintainer! Feel free to claim your Zephyr Maintainer badge here: https://forms.gle/oCw9iAPLhUsHTapc8 :)

**Pull request merged**